src/xrpld/app/misc/CredentialHelpers.cpp

```diff
@@ -225,6 +226,50 @@ makeSorted(STArray const& in)
     return out;
 }
 
+NotTEC
+checkArray(STArray const& credentials, unsigned maxSize, beast::Journal j)
+{
+    if (credentials.empty() || (credentials.size() > maxSize))
+    {
+        JLOG(j.trace()) << "Malformed transaction: "
+                           "Invalid credentials size: "
+                        << credentials.size();
+        return credentials.empty() ? temARRAY_EMPTY : temARRAY_TOO_LARGE;
+    }
```

Comment:
This feels easier to read/less confusing
```suggestion
    if (credentials.empty())
    {
        JLOG(j.trace()) << "Malformed transaction: Empty credentials list";
        return temARRAY_EMPTY;
    }
    if (credentials.size() > maxSize)
    {
        JLOG(j.trace()) << "Malformed transaction: "
                           "Invalid credentials size: "
                        << credentials.size();
        return temARRAY_TOO_LARGE;
    }
```

Reply:
For me it is more confusing and distracting to having almost the same blocks